tools/llvm-spirv/llvm-spirv.cpp

```diff
@@ -142,77 +144,114 @@ static std::string removeExt(const std::string &FileName) {
 
 static ExitOnError ExitOnErr;
 
-static int convertLLVMToSPIRV(const SPIRV::TranslatorOpts &Opts) {
-  LLVMContext Context;
+static void error(const Twine &Msg) {
+  errs() << "llvm-spirv: " << Msg << '\n';
+  exit(1);
+}
+
+static void error(std::error_code EC, const Twine &Prefix) {
+  if (EC)
+    error(Prefix + ": " + EC.message());
+}
+
+static void saveListToFile(std::string Filename, std::string Content) {
+  std::error_code EC;
+  raw_fd_ostream OS{Filename, EC, sys::fs::OpenFlags::OF_None};
+  error(EC, "error opening the file '" + Filename + "'");
+  OS.write(Content.data(), Content.size());
+}
 
-  std::unique_ptr<MemoryBuffer> MB =
-      ExitOnErr(errorOrToExpected(MemoryBuffer::getFileOrSTDIN(InputFile)));
-  std::unique_ptr<Module> M =
-      ExitOnErr(getOwningLazyBitcodeModule(std::move(MB), Context,
-                                           /*ShouldLazyLoadMetadata=*/true));
-  ExitOnErr(M->materializeAll());
+static void saveLLVMModuleToFile(std::string Filename, Module *M) {
+  std::error_code EC;
+  raw_fd_ostream OS{Filename, EC, sys::fs::OpenFlags::OF_None};
```

Comment:
Is this conformant with [do-not-use-braced-initializer-lists-to-call-a-constructor](https://llvm.org/docs/CodingStandards.html#do-not-use-braced-initializer-lists-to-call-a-constructor)?

Reply:
I can't say for sure since I didn't take a look into raw_fd_ostream constructor, but I was thinking about this initialization as an aggregate initialization and from LLVM Coding standards:

> use a braced initializer list (without any type for temporaries) when doing aggregate initialization or something notionally equivalent.